src/api/sandbox.rkt

```diff
@@ -277,34 +279,33 @@
       (set! timeline (*timeline*))
       (when seed
         (set-seed! seed))
-      (with-handlers ([exn? (curry on-exception start-time)])
-        (timeline-event! 'start) ; Prevents the timeline from being empty.
-        (define result
-          (match command
-            ['alternatives (get-alternatives test pcontext seed)]
-            ['evaluate (get-calculation test pcontext)]
-            ['cost (get-cost test)]
-            ['errors (get-errors test pcontext)]
-            ['exacts (get-exacts test pcontext)]
-            ['improve (get-alternatives/report test)]
-            ['local-error (get-local-error test pcontext)]
-            ['explanations (get-explanations test pcontext)]
-            ['sample (get-sample test)]
-            [_ (error 'compute-result "unknown command ~a" command)]))
-        (timeline-event! 'end)
-        (define time (- (current-inexact-milliseconds) start-time))
-        (job-result command test 'success time (timeline-extract) (warning-log) result))))
-
-  (define (in-engine _)
-    (if profile?
-        (profile-thunk (λ () (compute-result test))
-                       #:order 'total
-                       #:delay 0.01
-                       #:render (λ (p order) (write-json (profile->json p) profile?)))
-        (compute-result test)))
+      (define (go)
+        (with-handlers ([exn? (curry on-exception start-time)])
+          (timeline-event! 'start) ; Prevents the timeline from being empty.
+          (begin0 (match command
+                    ['alternatives (get-alternatives test pcontext seed)]
+                    ['evaluate (get-calculation test pcontext)]
+                    ['cost (get-cost test)]
+                    ['errors (get-errors test pcontext)]
+                    ['exacts (get-exacts test pcontext)]
+                    ['improve (get-alternatives/report test)]
+                    ['local-error (get-local-error test pcontext)]
+                    ['explanations (get-explanations test pcontext)]
+                    ['sample (get-sample test)]
+                    [_ (error 'compute-result "unknown command ~a" command)])
+            (timeline-event! 'end))))
+      (define result
```

Comment:
So calls to `improve` that fail or `timeout` return a job-result instead of a `improve-result` 

Matching on report side steps the underlying issue but allows the nightly to complete.
```
(match result
         [(? job-result?) result]
         [_
          (define time (- (current-inexact-milliseconds) start-time))
```
https://nightly.cs.washington.edu/reports/herbie/1733772136:nightly:zane-report-profile:4f4c955290/

Reply:
Good catch. I’ll fix it.